src/cli/commands/upgrade.js

```diff
@@ -27,16 +28,37 @@ export async function run(
     peerDependencies,
   } = await config.readRootManifest() || {};
   const allDependencies = Object.assign({}, peerDependencies, optionalDependencies, devDependencies, dependencies);
+  let addArgs = [];
 
-  const addArgs = args.map((dependency) => {
-    const remoteSource = allDependencies[dependency];
+  console.log(allDependencies);
 
-    if (remoteSource && PackageRequest.getExoticResolver(remoteSource)) {
-      return remoteSource;
+  if (flags.scope) {
+    const searchPattern = new RegExp(`^${flags.scope}`);
```

Comment:
1/ We probably want to run a check to make sure that `flags.scope` is a valid scope, and abort otherwise
2/ The regex above will make `-S @angular` upgrade `@angular`, but also `@angular-community` etc.